src/Screens/Main/Settings/UserAccount/MentorForm.tsx

```diff
@@ -39,6 +51,33 @@ export default ({ userId }: Props) => {
     }
   };
 
+  const changeStatusMessage = () => {
+    if (typeof mentor !== 'undefined') {
+      dispatch({
+        type: 'mentor/changeStatusMessage/start',
+        payload: { statusMessage, mentor },
+      });
+    }
+  };
+
+  const resetStatusMessage = () => {
+    dispatch({
+      type: 'mentor/changeStatusMessage/reset',
+      payload: undefined,
+    });
+  };
+
+  React.useEffect(() => {
+    if (RD.isSuccess(requestState)) {
+      const timeout = setTimeout(
+        resetStatusMessage,
+        changeStatusMessageState.coolDownDuration,
+      );
+
+      return () => clearTimeout(timeout);
+    }
+  }, [requestState]);
```

Comment:
Instead of a timeout, would it make sense to render an AlertBox that has option to push "OK" (reset the requestState) OR push "Try again" (make the request again)

But if we would keep this as a timeout, then only have the timeout and callback to reset the state in either in the UI or in the reducer =)

Reply:
That kind of AlertBox would make sense if the request has failed but probably not if it was successful?
Anyways, I agree that the timeout and reset should only happen in one place.

Reply:
Yes, if successfull, then it could go away by itself =)

Reply:
Well done Veera very nice code